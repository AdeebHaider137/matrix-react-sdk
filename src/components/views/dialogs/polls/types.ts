export type PollHistoryFilter = 'ACTIVE' | 'ENDED';