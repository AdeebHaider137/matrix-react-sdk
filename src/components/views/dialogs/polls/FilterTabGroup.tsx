/*
Copyright 2023 The Matrix.org Foundation C.I.C.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

import React, { FieldsetHTMLAttributes, ReactNode } from "react";

export type FilterTab<T> = {
    label: string | ReactNode;
    id: T;
};
type FilterTabGroupProps<T extends string = string> = FieldsetHTMLAttributes<any> & {
    name: string;
    onFilterChange: (id: T) => void;
    value: T;
    tabs: FilterTab<T>[];
};

export const FilterTabGroup = <T extends string = string>({
    name,
    value,
    tabs,
    onFilterChange,
    ...rest
}: FilterTabGroupProps<T>): JSX.Element => (
    <fieldset {...rest} className="mx_FilterTabGroup">
        {tabs.map(({ label, id }) => (
            <label key={id}>
                <input type="radio" name={name} value={id} onChange={() => onFilterChange(id)} checked={value === id} />
                {label}
            </label>
        ))}
    </fieldset>
);
