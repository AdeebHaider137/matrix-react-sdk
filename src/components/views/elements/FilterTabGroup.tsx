/*
Copyright 2023 The Matrix.org Foundation C.I.C.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

import React, { FieldsetHTMLAttributes, ReactNode } from "react";

export type FilterTab<T> = {
    label: string | ReactNode;
    id: T;
};
type FilterTabGroupProps<T extends string = string> = FieldsetHTMLAttributes<any> & {
    // group name used for radio buttons
    name: string;
    onFilterChange: (id: T) => void;
    // active tab's id
    value: T;
    // tabs to display
    tabs: FilterTab<T>[];
};

/**
 * Styles a set of content filters as tabs
 * Accepts type T for the group's value property
 * (example in Poll History)
 */
export const FilterTabGroup = <T extends string = string>({
    name,
    value,
    tabs,
    onFilterChange,
    ...rest
}: FilterTabGroupProps<T>): JSX.Element => (
    <fieldset {...rest} className="mx_FilterTabGroup">
        {tabs.map(({ label, id }) => (
            <label data-testid={`filter-tab-${name}-${id}`} key={id}>
                <input type="radio" name={name} value={id} onChange={() => onFilterChange(id)} checked={value === id} />
                <span>{label}</span>
            </label>
        ))}
    </fieldset>
);
