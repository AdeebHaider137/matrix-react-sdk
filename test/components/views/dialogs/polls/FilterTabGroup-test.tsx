
import React from 'react';
import { mount } from 'enzyme';

import { FilterTabGroup } from '../../../../../src/components/views/dialogs/polls/FilterTabGroup';

describe('<FilterTabSwitch />', () => {
    const defaultProps = {};
    const getComponent = (props = {}) =>
        mount(<FilterTabSwitch {...defaultProps} {...props} />);

    it('renders', () => {
        const component = getComponent();
        expect(component).toBeTruthy();
    });
});
