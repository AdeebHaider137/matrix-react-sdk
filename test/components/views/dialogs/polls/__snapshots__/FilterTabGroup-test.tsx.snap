// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterTabGroup /> renders options 1`] = `
<div>
  <fieldset
    class="mx_FilterTabGroup"
    data-testid="test"
  >
    <label>
      <input
        checked=""
        name="test"
        type="radio"
        value="Apple"
      />
      Label for Apple
    </label>
    <label>
      <input
        name="test"
        type="radio"
        value="Banana"
      />
      Label for Banana
    </label>
    <label>
      <input
        name="test"
        type="radio"
        value="Orange"
      />
      Label for Orange
    </label>
  </fieldset>
</div>
`;
